src/tests/src/libs/antares/study/series/timeseries-tests.cpp

```diff
@@ -0,0 +1,138 @@
+#define BOOST_TEST_MODULE "test time series"
+#define BOOST_TEST_DYN_LINK
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include <include/antares/series/series.h>
+
+using namespace Antares::Data;
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Fixture(const Fixture& f) = delete;
+    Fixture(const Fixture&& f) = delete;
+    Fixture& operator=(const Fixture& f) = delete;
+    Fixture& operator=(const Fixture&& f) = delete;
+    Fixture() : ts(tsnum)
+    {
+        ts.resize(1, HOURS_PER_YEAR);
+        tsnum.resize(1, 1);
+    }
+    TimeSeries ts;
+    TimeSeries::TSNumbers tsnum;
+    std::string folder;
+
+    void fillColumn(unsigned int idx);
+    void fillColumnReverse(unsigned int idx);
+
+    void fillTsnum();
+
+};
+
+void Fixture::fillColumn(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = i;
+}
+
+void Fixture::fillColumnReverse(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = HOURS_PER_YEAR - i;
+}
+
+void Fixture::fillTsnum()
+{
+    tsnum.resize(1, ts.timeSeries.width);
+    for (unsigned int i = 0; i < ts.timeSeries.width; i++)
+        tsnum[0][i] = i;
```

Comment:
This is bad test data:
the index and what it points to are the same, so we don't really test that the indirection works well.
Some explicit hard coded values would be better like `{3,0,2,1}` or whatever.